### PR TITLE
[Keyboard] New IDOBAO ID63 (a.k.a. Denwir D60)

### PR DESCRIPTION
**Pull request by vinorodrigues**
## Description

IDOBAO is proud to release the new IDOBAO ID63 to market - initial GB was via [Drop.com](https://drop.com/buy/idobao-id63-60-gasket-hot-swappable-aluminum-mechanical-keyboard-kit) *(April 2022)* with shipping expected around July 2022.

This board, a.k.a. "Denwir D60" or "D6", comes in two PCB options - this is the 63-key version *(with arrow keys)*.

The IDOBAO ID63 is a tailored gasket-mounted 60% keyboard with a unique build to an ultra-compact 63-key layout that includes arrow navigation keys.

### Additional items and notes in this PR

* Includes the factory default keymap in keymap folder `idobao`.  This adds 4-state RGB toggle key, and Caps-Lock LED indicator that is not part of the `default` and `via` keymaps.
* Default compile include all features of the PCB, but the IDOBAO ID63 cannot leverage underglow.  This can be disabled by compile command argument or "#define" as this PCB will be used for future case builds. *(the `idobao` keymap disables this by default)*

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

n/a -- *(New production keyboard)* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/idobao/id63/post_rules.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+#  this enables switching off thoes LEDs
+# Usage:  `make idobao/id63:default UNDERGLOW=off`
+
+ifeq ($(findstring off,$(UNDERGLOW)), off)
+  OPT_DEFS += -DID63_DISABLE_UNDERGLOW
+else ifeq ($(findstring no,$(UNDERGLOW)), no)
+  OPT_DEFS += -DID63_DISABLE_UNDERGLOW
+else ifeq ($(findstring 0,$(UNDERGLOW)), 0)
+  OPT_DEFS += -DID63_DISABLE_UNDERGLOW
+endif
```
Comment (drashna) [inline keyboards/idobao/id63/post_rules.mk:11]: This is ... a lot simpler. 
```suggestion
UNDERGLOW ?= yes
ifneq ($(strip $(UNDERGLOW)), yes)
  OPT_DEFS += -DID63_DISABLE_UNDERGLOW
endif
```

File: keyboards/idobao/id63/readme.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# IDOBAO ID63
+
+![IDOBAO ID63](https://idobao.github.io/assets/img/idobao-id63.png)
```
Comment (drashna) [inline keyboards/idobao/id63/readme.md:3]: please rehost this on Imgur

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/idobao/id63/post_rules.mk (hunk 1 above)
Comment (vinorodrigues) [reply keyboards/idobao/id63/post_rules.mk]: indeed ... much!
all recommendations adopted.

**Review by drashna — APPROVED** [review]

File: keyboards/idobao/id63/keymaps/idobao/config.h [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Copyright 2022 Vino Rodrigues (@vinorodrigues)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/* Denwir D60 has solid back case - no underglow LED's are visible. */
```
Comment (drashna) [inline keyboards/idobao/id63/keymaps/idobao/config.h:4]:
```suggestion
#pragma once

/* Denwir D60 has solid back case - no underglow LED's are visible. */
```

File: keyboards/idobao/id63/readme.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# IDOBAO ID63
+
+![IDOBAO ID63](https://i.imgur.com/DxlOr9x.png)
```
Comment (drashna) [inline keyboards/idobao/id63/readme.md:3]:
```suggestion
![IDOBAO ID63](https://i.imgur.com/DxlOr9xh.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**